boltops-tools/terraspace
# generator improvements: core examples for plugin=none. creates simple random_pet

This is a 🐞 bug fix.
This is a 🙋‍♂️ feature or enhancement.
This is a 🧐 documentation change.


## Summary

Fixes and improve generator so that `terraspace new project infra --plugin none --examples` option works right off the bat. This brings in non-cloud provider support as a 1st class citizen.

Also:

* cody project for plugin=none
* fix builds
* codebuild: use tfenv to install terraform
* autodetect plugin improvements
* add Terraspace.check_project for rspec-terraspace
* allow terraspace build to work when aws creds not properly setup. nil.to_s

## Context

* https://github.com/boltops-tools/terraspace/pull/174

## How to Test

    terraspace new project infra --plugin none --examples
    cd infra
    terraspace new test demo --type stack
    cd app/stacks/demo
    terraspace test

See: https://github.com/boltops-tools/terraspace/blob/master/.cody/none/bin/build.sh

## Version Changes

Minor
